spacy_llm/backends/rest/backend/base.py

```diff
@@ -69,31 +74,46 @@ def credentials(self) -> Dict[str, str]:
         """
 
     def retry(
-        self,
-        call_api: Callable[[], requests.Response],
+        self, call_method: Callable[..., requests.Response], url: str, **kwargs
     ) -> requests.Response:
         """Retry a call to an API if we get a non-ok status code.
         This function automatically retries a request if it catches a response
-        with an error code in `error_codes`. The amount of timeout also increases
+        with an error code in `error_codes`. The time interval also increases
         exponentially every time we retry.
-        call_api (Callable[[], requests.Response]): Call to API.
-        error_codes (Tuple[int]): Error codes indicating unsuccessful calls.
+        call_method (Callable[[str, ...], requests.Response]): Method to use to fetch request. Must accept URL as first
+            parameter.
+        url (str): URL to address in request.
+        kwargs: Keyword args to be passed on to request.
         RETURNS (requests.Response): Response of last call.
         """
-        timeout = self._timeout
-        response = call_api()
+
+        def _call_api() -> requests.Response:
+            """Calls API with given timeout.
+            RETURNS (requests.Response): Response object.
+            """
+            try:
+                return call_method(url, **kwargs)
+            except (ConnectTimeout, ReadTimeout, TimeoutError) as err:
```

Comment:
I'm confused, do we catch anything? These are immediately re-raised...

Reply:
Here we just add a more concise error message on top of the stack trace for easier error identification.

Reply:
Ah, right. I thought we'd want to retry after a timeout.

Reply:
Updated in https://github.com/explosion/spacy-llm/pull/110/commits/d97c3c6e076b727a3ca7cf0c4b11c3b64c20cac6. Feel free to resolve this if you think this is sufficient.

Reply:
Looks great!